smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/model/FlattenServiceErrorsOntoOperations.kt

```diff
@@ -0,0 +1,29 @@
+package software.amazon.smithy.swift.codegen.model
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.knowledge.TopDownIndex
+import software.amazon.smithy.model.shapes.ServiceShape
+
+object FlattenServiceErrorsOntoOperations {
+    /**
+     * Puts error structures defined at service level to every operation under that service.
+     * Pre-processing is required to ensure operations correctly model service errors in addition to operation errors.
+     */
+    fun transform(model: Model, service: ServiceShape): Model {
+        // Get service errors. If none exist, return unmodified model
+        val serviceErrors = if (service.errors == null || service.errors.isEmpty()) return model else service.errors
+
+        val topDownIndex: TopDownIndex = TopDownIndex.of(model)
+        val operations = topDownIndex.getContainedOperations(service)
+        val modelBuilder: Model.Builder = model.toBuilder()
+
+        // Append service errors to each and every operation
+        for (op in operations) {
+            modelBuilder.addShape(
+                op.toBuilder().addErrors(serviceErrors).build()
+            )
+        }
+
+        return modelBuilder.build()
```

Comment:
This code can be made into a bit more idiomatic kotlin. 
1. we have a return hidden in line 14, generally this is not good we want to try to keep possible exit scenarios explicit and easy to find. One way to do this is to replace `if x return y else z` with kotlin's takeUnless function
2. kotlin's forEach is preferred over a for loop

Maybe we can try re-writing this to following:
```
fun transform(model: Model, service: ServiceShape): Model {
    // Get service errors if any exist
    service.errors?.takeUnless { it.isEmpty() }?.let { serviceErrors ->
        val topDownIndex: TopDownIndex = TopDownIndex.of(model)
        val operations = topDownIndex.getContainedOperations(service)
        val modelBuilder: Model.Builder = model.toBuilder()

        // Append service errors to each and every operation
        operations.forEach { op ->
            modelBuilder.addShape(
                op.toBuilder().addErrors(serviceErrors).build()
            )
        }

        return modelBuilder.build()
    } ?: model // If service.errors is null or empty, return unmodified model
}
```

Reply:
Changed style to be more idiomatic to Kotlin as suggested.